web/packages/teleport/src/services/agents/types.ts

```diff
@@ -35,7 +35,7 @@ export type UnifiedResource =
 
 export type UnifiedResourceKind = UnifiedResource['kind'];
 
-export type ResourcesResponse<T extends UnifiedResource> = {
+export type ResourcesResponse<T> = {
```

Comment:
Why remove this constraint?

Reply:
It was not necessary (we dindn't use it), and removing it made it easier to write tests (I would otherwise need to create objects that fully conform to one of our resource types, with all required fields).

Reply:
I'm a bit skeptical of the idea that "we didn't use it" -- even if removing it doesn't break anything, it still may represent an important semantic constraint for changes going forward.

cc @kimlisa can you double check this?

Reply:
i agree with isaiah... T seems too generic especially if we can be specific (i'm sorry to be bearer of bad news)

Reply:
OK, gonna revert it and return dummy data in unit tests.